pacman_mirrors/pacman_mirrors.py

```diff
@@ -165,64 +156,129 @@ def command_line_parse(self):
         if args.branch:
             self.config["branch"] = args.branch
 
-        if args.mirror_dir:
-            self.config["mirror_dir"] = args.mirror_dir
-            self.default_mirror_dir = self.config["mirror_dir"]
-
-        self.available_countries = sorted(
-            os.listdir(self.config["mirror_dir"]))
+        if args.timeout:
+            self.max_wait_time = args.timeout
 
-        if args.geoip:
-            self.geolocation = True
+        if args.quiet:
+            self.quiet = True
 
-        if args.country:
-            country = args.country.split(",")
-            if country == ["Custom"]:
-                self.config["only_country"] = country
-            elif country == ["all"]:
-                self.config["only_country"] = []
-            else:
-                try:
-                    self.validate_country_list(country,
-                                               self.available_countries)
-                    self.config["only_country"] = country
-                except argparse.ArgumentTypeError as err:
-                    parser.error(err)
+        if args.mirror_dir:
+            self.config["mirror_dir"] = args.mirror_dir
 
         if args.output:
             if args.output[0] == "/":
                 self.config["mirror_list"] = args.output
             else:
                 self.config["mirror_list"] = os.getcwd() + "/" + args.output
-            self.default_mirror_list = self.config["mirror_list"]
 
         if args.interactive:
             self.interactive = True
             if not os.environ.get("DISPLAY") or not GTK_AVAILABLE:
                 self.no_display = True
+        # geoip and country are mutually exclusive
+        if args.geoip:
+            self.geolocation = True
 
-        if args.timeout:
-            self.max_wait_time = args.timeout
+        if args.country and not args.geoip:
+            self.config["only_country"] = args.country.split(",")
 
```

Comment:
Before I joined the project the output of pacman-mirrors -h was this.
It states that -c is ignored if --geoip is supplied. I just moved the internal check so it occurs earlier
```
usage: pacman-mirrors [-h] [-g] [-m {rank,random}]
                      [-b {stable,testing,unstable}] [-c COUNTRY] [--geoip]
                      [-d PATH] [-o FILE] [-t SECONDS] [--no-update] [-i] [-v]
                      [-q] [-f DIGIT]

optional arguments:
  -h, --help            show this help message and exit
  -g, --generate        Generate mirrorlist
  -m, --method {rank,random}
                        Generation method
  -b, --branch {stable,testing,unstable}
                        Branch name
  -c, --country COUNTRY
                        Comma separated list of countries, from which mirrors
                        will be used
  --geoip               Get current country using geolocation. Ignored if
                        '-c/--country' is supplied
  -d, --mirror_dir PATH
                        Mirrors list path
  -o, --output FILE     Output file
  -t, --timeout SECONDS
                        Maximum waiting time for server response
  --no-update           Don't generate mirrorlist if 'NoUpdate = True' in the
                        configuration file
  -i, --interactive     Generate custom mirrorlist
  -v, --version         Print the pacman-mirrors version
  -q, --quiet           Quiet mode - less verbose output
```